### ISSUE TITLE
Launch stream automatically when it goes live

### ISSUE THREAD

**Issue by clayote** [streamlink/streamlink-twitch-gui#947]
### Checklist

- [X] I understand that **Streamlink Twitch GUI** is just a launcher for **Streamlink**
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink-twitch-gui/blob/master/CONTRIBUTING.md)
- [X] [I have checked the list of open and recently closed issues](https://github.com/streamlink/streamlink-twitch-gui/issues?q=is%3Aissue+sort%3Aupdated-desc)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink-twitch-gui/commits/master)

### Description

I'd like to set a stream to automatically open as soon as it goes live.

I wanted this feature when a stream I like stopped broadcasting due to a momentary internet outage. They were back quickly, but I missed a couple minutes because I happened not to refresh in time. I'd have preferred to tell Streamlink that I'm waiting for them to come back, so give me their stream as soon as it's available.

I think this could build on top of the existing functionality for sending a notification when a stream is live.

**Comment by bastimeyer** [comment]
Duplicate of #241, #418

As explained in these threads from six/seven years ago, this is unfeasible because of how the player gets launched by Streamlink and not Streamlink Twitch GUI. Automatically opening streams, either via a scheduling process of some kind or via a reconnect functionality, will cause more issues than such a feature would solve, so anything in regards to automatism when launching streams can't and won't be implemented. As you can also see, nothing was ever implemented in regards to a different player launch system since then, because it's a ton of work with little to no gain, and it won't fix the issue anyway.

These issues/requests came up during the time when I just only working on the Twitch GUI and wasn't involved at all with Livestreamer and then the Streamlink fork after Livestreamer died. This has obviously changed now and nowadays I'm the only one involved with any major/deeper changes to Streamlink. A proper "reconnect" feature would have to be implemented in Streamlink itself, and this is currently still blocked by the old CLI implementation from the Livestreamer days, and also how the player process is launched and how stream data is fed to the player, and it's unclear if this issue can ever be resolved. This is not a trivial thing to fix, and very much related to handling stream discontinuities (ads, multiple/follow-up streams, etc.). Simply launching the player again is the same issue as described in the linked issues, and on top if it, it's a terrible user experience.

And as mentioned in the meta thread of the 2.0.0 release from last year where I had to re-implement the entire Twitch API stuff in order to keep the Twitch GUI alive (#849), my motivation for working on the Twitch GUI has been fading more and more over the years, especially with all the tech debt that has accumulated. The project has thus been in maintenance-mode for the past years and only simple features will get implemented, if it's reasonable and if I'm motivated enough. I don't expect any contributions from other people due to the outdated and unpopular tech being used here.

**Comment by invalidhippo** [comment]
For anyone interested, I have written an application that might offer such functionality.
